src/main/java/org/opentripplanner/routing/api/request/refactor/request/NewRouteRequest.java

```diff
@@ -0,0 +1,470 @@
+package org.opentripplanner.routing.api.request.refactor.request;
+
+import static org.opentripplanner.util.time.DurationUtils.durationInSeconds;
+
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Locale;
+import org.apache.commons.lang3.tuple.Pair;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.opentripplanner.api.common.LocationStringParser;
+import org.opentripplanner.common.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.model.GenericLocation;
+import org.opentripplanner.model.plan.SortOrder;
+import org.opentripplanner.model.plan.pagecursor.PageCursor;
+import org.opentripplanner.model.plan.pagecursor.PageType;
+import org.opentripplanner.routing.api.request.RoutingRequest;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.api.request.refactor.preference.RoutingPreferences;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.spt.DominanceFunction;
+import org.opentripplanner.routing.vehicle_rental.RentalVehicleType;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.util.time.DateUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: 2022-08-18 rename class when done
+public class NewRouteRequest implements Cloneable, Serializable {
```

Comment:
Is it OK if we keep it as it is for the moment? Moving RoutingPreferences into RoutingRequest would take some time since it is used in a lot of places. Maybe we can take care of that after we merge?